discovery_child_development/notebooks/openalex/06_fine_tune_embeddings.py

```diff
@@ -319,5 +346,62 @@ def load_trainer(
 compute_metrics(predictions)
 
 # %%
+predictions
+
+# %%
+y_pred = binarise_predictions(predictions.predictions, threshold=0.5)
+y_pred_proba = torch.sigmoid(torch.Tensor(predictions.predictions)).numpy()
+y_true = predictions.label_ids
+
+# %%
+label_names = list(Y_val.columns[1:])
+
+# %%
+from sklearn.metrics import precision_score, recall_score
+
+precision = precision_score(y_true, y_pred, average="macro")
+recall = recall_score(y_true, y_pred, average="macro")
+print(f"Precision: {precision}; recall: {recall}")
+
+# %%
+# get the metrics for individual labels
+precision = precision_score(y_true, y_pred, average=None)
+recall = recall_score(y_true, y_pred, average=None)
+print(f"Precision: {precision}; recall: {recall}")
+
+# %%
+classification_utils.evaluate_model_performance(
+    y_true, y_pred, y_pred_proba, label_names
+)
+
+# %%
+confusion_matrix = classification_utils.create_confusion_matrix(
+    y_true, y_pred, label_names, proportions=False
+)
+
+# %%
+classification_utils.create_heatmap_table(
+    y_true, y_pred, label_names, proportions=False
+)
+
+# %%
+metrics = classification_utils.create_average_metrics(y_true, y_pred, average="macro")
+metrics
+
+# %%
+wandb.run.summary["macro_avg_f1"] = metrics["f1"]
+wandb.run.summary["accuracy"] = metrics["accuracy"]
+wandb.run.summary["macro_avg_precision"] = metrics["precision"]
+wandb.run.summary["macro_avg_recall"] = metrics["recall"]
+
+# %%
+# Log confusion matrix
+wb_confusion_matrix = wandb.Table(
+    data=confusion_matrix, columns=confusion_matrix.columns
+)
+run.log({"confusion_matrix": wb_confusion_matrix})
+
+# %%
+wandb.finish()
```

Comment:
Was having a look at your previous wandb run and our accuracy is a little different? Is that to be expected with the hugging face models or should it be the same since you specified the seed?

Reply:
🧐 no clue. Will have a google